__tests__/github.test.js

```diff
@@ -0,0 +1,57 @@
+const pool = require('../lib/utils/pool');
+const setup = require('../data/setup');
+const request = require('supertest');
+const app = require('../lib/app');
+const UserService = require('../lib/services/UserService');
+jest.mock('../lib/utils/github');
+
+describe('gitty routes', () => {
+  beforeEach(() => {
+    return setup(pool);
+  });
+
+  afterAll(() => {
+    pool.end();
+  });
+
+  it('redirects to the github oauth page upon login', async () => {
+    const req = await request(app).get('/api/v1/github/login');
+    expect(req.header.location).toMatch(
+      /https:\/\/github.com\/login\/oauth\/authorize\?client_id=[\w\d]+&scope=user&redirect_uri=http:\/\/localhost:7890\/api\/v1\/github\/login\/callback/i
+    );
+  });
+
+  it('redirects user to dashboard after successful login', async () => {
```

Comment:
No dashboard required for this particular deliverable. The redirect was supposed to go straight to /api/v1/posts :)